steps.tf

```diff
@@ -26,14 +26,30 @@ resource "aws_s3_bucket_object" "generate_analytical_dataset_script" {
   )
 }
 
+resource "aws_s3_bucket_object" "generate_dataset_from_htme_script" {
+  bucket = data.terraform_remote_state.common.outputs.config_bucket.id
+  key    = "component/analytical-dataset-generation/generate_dataset_from_htme.py"
+  content = templatefile("${path.module}/steps/generate_dataset_from_htme.py",
+  {
+    secret_name        = local.secret_name
+    published_db       = "analytical_dataset_generation"
+    file_location      = "analytical-dataset"
+    url                = format("%s/datakey/actions/decrypt", data.terraform_remote_state.crypto.outputs.dks_endpoint[local.environment])
+    aws_default_region = "eu-west-2"
+    log_path           = "/var/log/adg/generate-analytical-dataset.log"
+    s3_prefix          = "businessdata/mongo/ucdata/2020-07-26"
+  }
+  )
+}
+
 resource "aws_s3_bucket_object" "hive_setup_sh" {
   bucket = data.terraform_remote_state.common.outputs.config_bucket.id
   key    = "component/analytical-dataset-generation/hive-setup.sh"
   content = templatefile("${path.module}/steps/hive-setup.sh",
     {
       hive-scripts-path           = format("s3://%s/%s", data.terraform_remote_state.common.outputs.config_bucket.id, aws_s3_bucket_object.create-hive-tables.key)
       python_logger               = format("s3://%s/%s", data.terraform_remote_state.common.outputs.config_bucket.id, aws_s3_bucket_object.logger.key)
-      generate_analytical_dataset = format("s3://%s/%s", data.terraform_remote_state.common.outputs.config_bucket.id, aws_s3_bucket_object.generate_analytical_dataset_script.key)
+      generate_analytical_dataset = format("s3://%s/%s", data.terraform_remote_state.common.outputs.config_bucket.id, aws_s3_bucket_object.generate_dataset_from_htme_script.key)
```

Comment:
These can be changed to reference the bucket name from the aws_s3_bucket_object themselves, instead of depending on the remote state. e.g. `format("s3://%s/%s", aws_s3_bucket_object.generate_dataset_from_htme_script.bucket, aws_s3_bucket_object.generate_dataset_from_htme_script.key)`